homeassistant/components/media_player/heos.py

```diff
@@ -0,0 +1,217 @@
+"""Denon HEOS Media Player."""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (PLATFORM_SCHEMA,
+                                                   MediaPlayerDevice)
+
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_MUSIC, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PLAY,
+    SUPPORT_PLAY_MEDIA, SUPPORT_PREVIOUS_TRACK, SUPPORT_SEEK, SUPPORT_STOP,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET, SUPPORT_VOLUME_STEP)
+
+from homeassistant.const import (CONF_HOST, CONF_NAME, CONF_PASSWORD,
+                                 CONF_USERNAME, STATE_OFF, STATE_PAUSED,
+                                 STATE_PLAYING, STATE_UNKNOWN)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['aioheos==0.1.5']
+
+DEFAULT_NAME = 'HEOS Player'
+
+SUPPORT_HEOS = SUPPORT_PLAY | SUPPORT_STOP | SUPPORT_PAUSE | \
+        SUPPORT_PLAY_MEDIA | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+        SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_STEP | \
+        SUPPORT_SEEK
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST):
+    cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME):
+    cv.string,
+    vol.Optional(CONF_USERNAME):
```

Comment:
It's not clear to me why we ask for username and password...  It's not needed for operating the device.  It's reasonable to me that users should have already logged the device into their account, otherwise we show it as unavailable.